tests/test_updater.py

```diff
@@ -0,0 +1,253 @@
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+    # Reset the "check" field and check enabling update checks.
+    updater.dangerzone.settings.updater_set("check", None)
+    alert_mock().launch.return_value = True
+    expected_settings["check"] = True
+    assert updater.should_check_for_updates() == False
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+    # Third run
+    #
+    # From the third run onwards, users should never be prompted for enabling update
+    # checks.
+    alert_mock.side_effect = RuntimeError("Should not be called")
+    for check in [True, False]:
+        updater.dangerzone.settings.updater_set("check", check)
+        assert updater.should_check_for_updates() == check
+
+
+def test_update_checks(
+    updater: Updater, monkeypatch: MonkeyPatch, mocker: MockerFixture
+) -> None:
+    """Test version update checks."""
+    # This dictionary will simulate GitHub's response.
+    mock_upstream_info = {"tag_name": f"v{get_version()}", "body": "changelog"}
+
+    # Make requests.get().json() return the above dictionary.
+    requests_mock = mocker.MagicMock()
+    requests_mock().json.return_value = mock_upstream_info
+    monkeypatch.setattr(updater_mod.requests, "get", requests_mock)
+
+    # Test 1 - Check that the current version triggers no updates.
+    report = updater.check_for_updates()
+    assert report == {}
+
+    # Test 2 - Check that a newer version triggers updates, and that the changelog is
+    # rendered from Markdown to HTML.
+    mock_upstream_info["tag_name"] = "v99.9.9"
+    report = updater.check_for_updates()
+    assert report == {
+        "version": "99.9.9",
+        "changelog": "<p>changelog</p>",
+        "error": None,
+    }
+
+    # Test 3 - Check that HTTP errors are converted to error reports.
+    requests_mock.side_effect = Exception("failed")
+    report = updater.check_for_updates()
+    assert report == {
+        "error": "failed",
+    }
+
+    # Test 4 - Check that cached version/changelog info do not trigger an update check.
+    updater.dangerzone.settings.updater_set("latest_version", "99.9.9")
+    updater.dangerzone.settings.updater_set("latest_changelog", "<p>changelog</p>")
+
+    report = updater.check_for_updates()
+    assert report == {
+        "version": "99.9.9",
+        "changelog": "<p>changelog</p>",
+        "error": None,
+    }
+
+
+def test_qt(
+    qtbot: QtBot, qt_updater: Updater, monkeypatch: MonkeyPatch, mocker: MockerFixture
+) -> None:
+    updater = qt_updater
+    updater.dangerzone.settings.updater_set("check", True)
+    updater.dangerzone.settings.updater_set("first_run", False)
+    expected_settings = default_updater_settings()
+    expected_settings["first_run"] = False
+    expected_settings["check"] = True
+
+    mock_upstream_info = {"tag_name": f"v{get_version()}", "body": "changelog"}
+
+    # Make requests.get().json() return the above dictionary.
```

Comment:
Just so we don't forget:
```suggestion
    # FIXME Make requests.get().json() return the above dictionary.
```

Reply:
Uhm, this is not a FIXME item actually. The mocking does in fact make `requests.get().json()` return the specified dict.

Reply:
OK